solver/pb/ops.proto

```diff
@@ -211,4 +199,67 @@ message Definition {
 message HostIP {
 	string Host = 1;
 	string IP = 2;
+}
+
+message FileOp {
+	repeated FileAction actions = 2;
+}
+
+message FileAction {
```

Comment:
Many of the fields in the messages added here could use some commentary to explain their behaviours/constraints etc.